2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -177,21 +177,92 @@ function run_mongod_supervisor() {
   ${CONTAINER_SCRIPTS_PATH}/replica_supervisor.sh 2>&1 &
 }
 
-# mongo_create_users creates the MongoDB admin user and the database user
-# configured by MONGODB_USER
-function mongo_create_users() {
-  mongo ${MONGODB_DATABASE} --eval "db.addUser({user: '${MONGODB_USER}', pwd: '${MONGODB_PASSWORD}', roles: [ 'readWrite' ]})"
+# mongo_create_admin creates the MongoDB admin user with password: MONGODB_ADMIN_PASSWORD
+# $1 - login parameters for mongo (optional)
+# $2 - host where to connect (localhost by default)
+function mongo_create_admin() {
+  if [[ -z "${MONGODB_ADMIN_PASSWORD:-}" ]]; then
+    echo "=> MONGODB_ADMIN_PASSWORD is not set. Authentication can not be set up."
+    exit 1
+  fi
+
+  # Set admin password
+  set +e
+  mongo admin ${1:-} --host ${2:-"localhost"} --eval "db.addUser({user: 'admin', pwd: '${MONGODB_ADMIN_PASSWORD}', roles: ['dbAdminAnyDatabase', 'userAdminAnyDatabase' , 'readWriteAnyDatabase','clusterAdmin' ]});"
+  local result=$?
+  set -e
+
+  if [[ ${result} -ne 0 ]]; then
+    echo "=> Failed to create MongoDB admin user."
+    exit 1
+  fi
+
+  touch ${MONGODB_DATADIR}/.mongodb_datadir_initialized
```

Comment:
BTW, another tiny benefit of creating `.mongodb_datadir_initialized` file outside of this function: we don't use this file in replica set configuration. But with old approach we've always created it.